authentication/php/phpleague/src/Oauth/UsernamePasswordContext.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.
+ * Licensed under the MIT License.  See License in the project root
+ * for license information.
+ */
+
+
+namespace Microsoft\Kiota\Authentication\Oauth;
+
+/**
+ * Class UsernamePasswordContext
+ *
+ * Params for the Resource Owner Password Credentials flow using a secret
+ *
+ * @package Microsoft\Kiota\Authentication
+ * @copyright 2022 Microsoft Corporation
+ * @license https://opensource.org/licenses/MIT MIT License
+ * @link https://developer.microsoft.com/graph
+ */
+class UsernamePasswordContext extends BaseSecretContext
+{
+    /**
+     * @var string
+     */
+    private string $username;
+    /**
+     * @var string
+     */
+    private string $password;
+    /**
+     * @var array
+     */
+    private array $additionalParams;
+
+    /**
+     * @param string $tenantId
+     * @param string $clientId
+     * @param string $clientSecret
+     * @param string $username
+     * @param string $password
+     * @param array $additionalParams
+     */
+    public function __construct(string $tenantId, string $clientId, string $clientSecret, string $username, string $password, array $additionalParams = [])
```

Comment:
I'm not sure we want to implement that flow. It'll fail in case MFA is setup, it's a security risk as well. Azure identity has something related but MSAL does not have it on purpose, lots of bad history here...

Reply:
Tagging you on a recent comment on the initial design doc.

Reply:
Thanks for feedback on the doc. Removed this flow